tools/pd-ctl/pdctl/ctl.go

```diff
@@ -26,51 +26,26 @@ import (
 	"github.com/tikv/pd/tools/pd-ctl/pdctl/command"
 )
 
-// CommandFlags are flags that used in all Commands
-type CommandFlags struct {
-	URL      string
-	CAPath   string
-	CertPath string
-	KeyPath  string
-	Help     bool
-}
-
 var (
-	commandFlags = CommandFlags{
-		URL: "http://127.0.0.1:2379",
-	}
-
-	detach            bool
-	interact          bool
-	version           bool
 	readlineCompleter *readline.PrefixCompleter
 )
 
 func init() {
 	cobra.EnablePrefixMatching = true
 }
 
-func pdctlRun(cmd *cobra.Command, args []string) {
-	if version {
-		server.PrintPDInfo()
-		return
-	}
-	if interact {
-		loop()
-	}
-}
-
-func getBasicCmd() *cobra.Command {
+// GetRootCmd is exposed for integration tests. But it can be embedded into another suite, too.
+func GetRootCmd() *cobra.Command {
 	rootCmd := &cobra.Command{
 		Use:   "pd-ctl",
 		Short: "Placement Driver control",
 	}
 
-	rootCmd.PersistentFlags().StringVarP(&commandFlags.URL, "pd", "u", commandFlags.URL, "address of pd")
-	rootCmd.PersistentFlags().StringVar(&commandFlags.CAPath, "cacert", commandFlags.CAPath, "path of file that contains list of trusted SSL CAs")
-	rootCmd.PersistentFlags().StringVar(&commandFlags.CertPath, "cert", commandFlags.CertPath, "path of file that contains X509 certificate in PEM format")
-	rootCmd.PersistentFlags().StringVar(&commandFlags.KeyPath, "key", commandFlags.KeyPath, "path of file that contains X509 key in PEM format")
-	rootCmd.PersistentFlags().BoolVarP(&commandFlags.Help, "help", "h", false, "help message")
+	rootCmd.PersistentFlags().StringP("pd", "u", "http://127.0.0.1:2379", "address of pd")
+	rootCmd.PersistentFlags().String("cacert", "", "path of file that contains list of trusted SSL CAs")
+	rootCmd.PersistentFlags().String("cert", "", "path of file that contains X509 certificate in PEM format")
+	rootCmd.PersistentFlags().String("key", "", "path of file that contains X509 key in PEM format")
+	rootCmd.PersistentFlags().BoolP("help", "h", false, "help message")
```

Comment:
unused?

Reply:
Yes. I found that cobra will add a `-h` anyway even if you did not provide one. The only difference is that cobra shows `help for ${argv[0]}`. It is removed.